### PR TITLE
Cohorts now can have a name.

### PR DESCRIPTION
**Pull request by PBara7a**
https://github.com/boolean-uk/team-dev-server/issues/119

### CODE REVIEW

**Comment by PBara7a** [comment]
One item from the issue checklist is "Have handled all error cases (try/catch, checked user input, etc.)".
Instead of checking the type of input, I converted it into a String before attempting to create a new record of Cohort. (domain/cohort.js, line 8)
Is this a valid approach? Or is it better to throw an error if the user does not provide a String as the cohort name?

File: src/domain/cohort.js [hunk 1]
```diff
@@ -4,12 +4,20 @@ import dbClient from '../utils/dbClient.js'
...
-  })
+export async function createCohort(cohortName) {
+  cohortName = String(cohortName)
 
-  return new Cohort(createdCohort.id)
+  let createdCohort
+
+  try {
+    createdCohort = await dbClient.cohort.create({
+      data: { cohortName: cohortName || '' }
```
Comment (dearshrewdwit) [inline src/domain/cohort.js:14]: Let's also remove the default value for a cohort name - throw an error if it's anything but a string of 1 or more chars.

File: src/domain/cohort.js [hunk 2]
```diff
@@ -4,12 +4,20 @@ import dbClient from '../utils/dbClient.js'
...
-  const createdCohort = await dbClient.cohort.create({
-    data: {}
-  })
+export async function createCohort(cohortName) {
+  cohortName = String(cohortName)
 
-  return new Cohort(createdCohort.id)
+  let createdCohort
+
+  try {
```
Comment (dearshrewdwit) [inline src/domain/cohort.js:12]: remove the try/ catch block

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/controllers/cohort.js [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
 
 export const create = async (req, res) => {
+  const { cohort_name: cohortName } = req.body
+
   try {
-    const createdCohort = await createCohort()
+    const createdCohort = await createCohort(cohortName)
+
+    if (!(createdCohort instanceof Cohort)) {
```
[the remark this answers is not in the document]
Comment (PBara7a) [reply src/controllers/cohort.js]: If statement was removed

File: src/domain/cohort.js (hunk 1 above)
Comment (PBara7a) [reply src/domain/cohort.js]: Error thrown if String has less than 1 characters.

File: src/domain/cohort.js (hunk 2 above)
Comment (PBara7a) [reply src/domain/cohort.js]: Removed.

File: src/domain/cohort.js [hunk 4]
```diff
@@ -4,12 +4,20 @@ import dbClient from '../utils/dbClient.js'
  * Create a new Cohort in the database
  * @returns {Cohort}
  */
-export async function createCohort() {
+export async function createCohort(cohortName) {
+  if (typeof cohortName !== 'string') {
```
Comment (dearshrewdwit) [inline src/domain/cohort.js:8]: The next stage of this is to read through https://www.prisma.io/docs/concepts/components/prisma-client/custom-validation

And use a validation library (prisma doesn't feature it, but supports 3rd party libraries)

**Review by dearshrewdwit — DISMISSED** [review]
Looks good to me - though the linter has picked up some violations

**Review by dearshrewdwit — APPROVED** [review]
LGTM

**Review by Alex90Jennings — APPROVED** [review]

**Pull request merged**